### PR TITLE
WIP - feature/new project ui

### PR DESCRIPTION
**Pull request by MutazAshhab**
Frames 102 and 99 from https://www.figma.com/file/vPh49OC7hJpMD7cQMAnBUo/UI-Design?node-id=416%3A2134

Includes:
- Close button on the top of every modal actually closes the modal
- The back button now takes you back a step rather than closing the modal, however, it will not retain the project name.
- after clicking transcribe you'll navigate to the old transcription UI.
- `continue` and `transcribe` are disabled when the relevant data in not yet uploaded/typed in

What has not been included in this PR:
- Delete button functionality (now does nothing when you click it)
- The text color is black (couldnt get it to change to white, not kidding, :( )
- Drag and drop is not implemented.
- The keyboard key `Return` and does not navigate to the next view

### notes for reviewers
- my css skills are whack so if something could be done in a simpler way bring it up.
- i have fallen in love with the `Stack` component and might have overused it. so again, bring it up in cases were we dont need stacks
- in the 2nd screenshot, you can see there is an orange outline, thats only there when i use the keyboard outside of the textfield and when you dont click on it.
- last note. im sorry for the 478 line code review that you'll have to go through

Screenshots:

image

image

image

image

### CODE REVIEW

**Comment by MutazAshhab** [comment]
As of [fbd4db85](https://github.com/patrickbrett/mlvet/pull/38/commits/fd4db85c62683af57e51e0ad2c503a9d9232a13d) there are still things that need to be done.

Things to be done:
- [x] Update the height and width of the modal container to match the designs on figma.
- [x] Fixing up the styles (including the white/black coloring of text and small parts of components.)
- [ ] Drag and drop.

**Comment by chloebrett** [comment]
Tested out the UI locally, looking good so far!

File: src/renderer/components/ActionButton.tsx
Comment (emmapainter) [inline src/renderer/components/ActionButton.tsx]: This button style should be called PrimaryButton

File: src/renderer/components/CancelButton.tsx
Comment (emmapainter) [inline src/renderer/components/CancelButton.tsx]: This should be SecondaryButton

File: src/renderer/components/ModalTitle.tsx
Comment (emmapainter) [inline src/renderer/components/ModalTitle.tsx]: The title should use h1 font style

File: src/renderer/components/NewProjectModal/ModalContainer.tsx
Comment (emmapainter) [inline src/renderer/components/NewProjectModal/ModalContainer.tsx]: Not sure on this (@hhen0005 help) but I think the modal should be constant width - currently the modal scales poorly for smaller window sizes.

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (emmapainter) [inline src/renderer/components/NewProjectModal/NewProjectView.tsx]: Do you know why these button labels are displaying in all caps?

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (emmapainter) [inline src/renderer/components/NewProjectModal/NewProjectView.tsx]: Figuring out how to change the non-focus and input text colours for this is pretty crucial - the current contrast is really hard to see.

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (emmapainter) [inline src/renderer/components/NewProjectModal/NewProjectView.tsx]: Would love it if the buttons were equal width and took up the width of the modal like this. Also could you add the drop shadows?
Screen Shot 2022-05-05 at 1 21 58 pm

File: src/renderer/components/SelectMediaBlock.tsx
Comment (emmapainter) [inline src/renderer/components/SelectMediaBlock.tsx]: @hhen0005 / @fwang1703 / someone experienced with front end - the border colour is set to grey 500.... so why does it show up as black in the app?

File: src/renderer/components/SelectMediaBlock.tsx
Comment (emmapainter) [inline src/renderer/components/SelectMediaBlock.tsx]: I know this colour isn't even showing up as white but the text colour here is grey 300

File: src/renderer/components/SelectMediaBlock.tsx
Comment (emmapainter) [inline src/renderer/components/SelectMediaBlock.tsx]: This text should use p-300 style

File: src/renderer/components/NewProjectModal/UploadVideoView.tsx
Comment (emmapainter) [inline src/renderer/components/NewProjectModal/UploadVideoView.tsx]: The location of this is independent of other elements - on smaller screen sizes it glitches out and displays over the top of the select media block
Screen Shot 2022-05-05 at 1 55 29 pm

**Review by emmapainter — CHANGES_REQUESTED** [review]
Added a few UI comments - also not sure if this is expected behaviour, but if you exit out from the select media modal (via the x or clicking outside of modal bounds) then click new project again it will take you back to the select media modal instead of the project name modal. When you're doing UI stuff it's worth testing it at a smaller window size (even just laptop screen size) to make sure everything still looks as expected.

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: Nope, but its probably a setting from the UI library

File: src/renderer/components/ActionButton.tsx
Comment (ghost) [reply src/renderer/components/ActionButton.tsx]: We shouldn't need to define primary and secondary button components. 
Because of the custom theme that we are using globally, if we define the color prop of a button as 'primary' or 'secondary' according to the MUI docs, then it should automatically already be styled according to our designs.

Hence, we don't actually need to do any other styling.

Comment (ghost) [reply src/renderer/components/ActionButton.tsx]: Now realising, theme is currently not working. Please use the method described above for now and I'll fix the theme.

Comment (MutazAshhab) [reply src/renderer/components/ActionButton.tsx]: I changed the functionality that is executed when the CloseIcon is clicked. It now closes the modal and resets back to the first view.

File: src/renderer/components/SelectMediaBlock.tsx
Comment (chloebrett) [reply src/renderer/components/SelectMediaBlock.tsx]: MUI really do be like that

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (chloebrett) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: Yeah it's a default part of the button styling - they have `text-transform` applied

**Comment by chloebrett** [comment]
Emma has made some good points, I'll wait until you've addressed them before reviewing

File: src/renderer/components/NewProjectModal/ModalContainer.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/ModalContainer.tsx]: Resolved in this commit: https://github.com/patrickbrett/mlvet/pull/38/commits/903905ef5e1001fef54cf9b154e015b8835560ac

File: src/renderer/components/CancelButton.tsx
Comment (MutazAshhab) [reply src/renderer/components/CancelButton.tsx]: Switched to standard button in: https://github.com/patrickbrett/mlvet/pull/38/commits/664fdb48cc831149ec5dc99c4985f94fff44bf5e

File: src/renderer/components/SelectMediaBlock.tsx
Comment (MutazAshhab) [reply src/renderer/components/SelectMediaBlock.tsx]: Missing semicolon on line 13, this is now fixed

File: src/renderer/components/NewProjectModal/UploadVideoView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/UploadVideoView.tsx]: Responsiveness is now fixed in: https://github.com/patrickbrett/mlvet/pull/38/commits/903905ef5e1001fef54cf9b154e015b8835560ac

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (emmapainter) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: Figured this out - in themes.tsx under typography -> button add `textTransform: 'none',`

File: src/renderer/components/NewProjectModal/UploadVideoView.tsx
Comment (ghost) [inline src/renderer/components/NewProjectModal/UploadVideoView.tsx]: Nice work defining the custom stack! All 4 of these properties can actually go in the custom stack styled part.
Off the top of my head:
direction = flex-direction
alignItems = align-items
justifyContent = justify-content
height = height

File: src/renderer/components/NewProjectModal/UploadVideoView.tsx
Comment (ghost) [inline src/renderer/components/NewProjectModal/UploadVideoView.tsx]: same as above :)

File: src/renderer/components/SelectMediaBlock.tsx
Comment (MutazAshhab) [reply src/renderer/components/SelectMediaBlock.tsx]: Fixed, again, it was the semicolon issue.

File: src/renderer/components/ModalTitle.tsx
Comment (MutazAshhab) [reply src/renderer/components/ModalTitle.tsx]: now using h-100, done in: https://github.com/patrickbrett/mlvet/pull/38/commits/edcab9ea48b51441c79e70ab36a1c046b144fcaa

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: Done in: https://github.com/patrickbrett/mlvet/pull/38/commits/9f8bba6985635aa8197025abb5cc262102356eb7

File: src/renderer/components/NewProjectModal/UploadVideoView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/UploadVideoView.tsx]: done in: https://github.com/patrickbrett/mlvet/pull/38/commits/9c3bb86622a4c497c326922f4841c70ba1e791ac

File: src/renderer/components/NewProjectModal/UploadVideoView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/UploadVideoView.tsx]: done in: https://github.com/patrickbrett/mlvet/pull/38/commits/9c3bb86622a4c497c326922f4841c70ba1e791ac

File: src/renderer/components/SelectMediaBlock.tsx
Comment (MutazAshhab) [reply src/renderer/components/SelectMediaBlock.tsx]: done in: https://github.com/patrickbrett/mlvet/pull/38/commits/724472e1594ea09a26fe28ffdedd877554a9e3ec

File: src/renderer/components/ActionButton.tsx
Comment (MutazAshhab) [reply src/renderer/components/ActionButton.tsx]: @hhen0005 update on this? currently im using Standard button and using color="primary"/"secondary". You mentioned something about variants? - sorry i forgot

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: Turns the color attribute in the TextField component does not take actual colors. Check the docs. https://mui.com/material-ui/api/text-field/

Color just changes the outline of the TextField and not the text color that the user types.

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: done in: https://github.com/patrickbrett/mlvet/pull/38/commits/ab49c26e4721cc25c00fd971de2e26f92a3ed5cd

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (MutazAshhab) [inline src/renderer/components/NewProjectModal/NewProjectView.tsx]: @hhen0005  @emmapainter When using the variant option, it gives us the correct size but it does not apply the correct coloring. Should that be applied separately or should it be done via the theme?

Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: Answered on zoom, coloring should be applied when Typography is used

File: src/main/util.ts
Comment (chloebrett) [inline src/main/util.ts]: Maybe return an enum instead, as this will always (or at least should, assuming the user is on a supported OS) have exactly one 'true' value

File: src/renderer/components/NewProjectModal/ModalContainer.tsx
Comment (chloebrett) [inline src/renderer/components/NewProjectModal/ModalContainer.tsx]: This was originally called `ImportMediaView`, for two reasons:
- It's not a really an upload, since there's no actual file transfer taking place - hence 'import' 
- It's not guaranteed to be a video file, it could be an audio file - hence 'media'

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (chloebrett) [inline src/renderer/components/NewProjectModal/NewProjectView.tsx]: In general it's a good idea to avoid abbreviations in variable names unless they're absolutely necessary - this in itself is fine but it's a slippery slope to obtuse names like 'genImpExpStats' that become meaningless (trust me I've seen it happen 😅)

File: src/renderer/pages/Project.tsx
Comment (chloebrett) [inline src/renderer/pages/Project.tsx]: IIRC the 'StandardButton' thing you deleted was made by Hugh as a placeholder for when we want to bring in custom styling. I'm fine with replacing with a regular MUI button for now as we can style it in the theme file instead (and currently have no custom styles for it), but maybe check with @hhen0005 and see what he thinks (that said I don't think this is a blocker for the merge)

File: src/renderer/pages/Project.tsx
Comment (patrickbrett) [inline src/renderer/pages/Project.tsx:92]: Again not a blocker, but if you have a minute can you remove the undo/redo buttons please as their functionality exists in the menu now (I know this is technically not part of your PR but just noticed it)

File: src/renderer/util.ts
Comment (chloebrett) [inline src/renderer/util.ts:23]: I think this should be a backend function, that way the renderer doesn't need to care about OS stuff. We might need a getOS() down the line but we're trying to make the platforms have feature parity so I can't foresee anything specific

File: src/renderer/util.ts
Comment (chloebrett) [inline src/renderer/util.ts:128]: Nice way of splitting it!

File: src/sharedTypes.ts
Comment (chloebrett) [inline src/sharedTypes.ts]: Same idea, if we do this on the back end then no shared types needed (also again, would make this an enum)

**Review by chloebrett — CHANGES_REQUESTED** [review]
Styling is looking really good! Just a few back end comments

File: src/renderer/pages/Project.tsx
Comment (emmapainter) [reply src/renderer/pages/Project.tsx]: Can confirm we will be removing the StandardButton

**Comment by emmapainter** [comment]
There is extra space at the top and bottom of the new project modal and at the top of the upload media modal - otherwise the styling is looking great! Sorry for being so picky 😅
image
On Figma if you select an element then select "inspect" on the top right then it will show you the pixel distance between the item you have selected and any item you hover over with your mouse
Screen Shot 2022-05-07 at 4 27 24 pm

**Comment by MutazAshhab** [comment]
Here is what it looks like now: 

image

corresponding commit: https://github.com/patrickbrett/mlvet/pull/38/commits/0f0ae23e5d6d70f8488f0a84e7c266b344c1d756

File: src/renderer/components/NewProjectModal/ModalContainer.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/ModalContainer.tsx]: refactor done in: https://github.com/patrickbrett/mlvet/pull/38/commits/40010a865dd4a23174225bea93e2d2876e78655d

File: src/renderer/util.ts
Comment (MutazAshhab) [reply src/renderer/util.ts:128]: thank you!

File: src/renderer/components/NewProjectModal/NewProjectView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/NewProjectView.tsx]: Ah yes, this is something i overlooked before pushing, there is no need for an abbreviated name here. I refactored the name here: https://github.com/patrickbrett/mlvet/pull/38/commits/5545d5ec5937464dc3b775d37bfe19d4e5b354a6

Although its fine to have an abbreviated variable name, changing this just so others dont see an abbreviated variable name and start getting ideas.

File: src/renderer/pages/Project.tsx
Comment (MutazAshhab) [reply src/renderer/pages/Project.tsx]: I have already removed all occurrences of StandardButton in this branch. Once it gets merged, `develop` will no longer have StandardButton and will only use Button.

File: src/sharedTypes.ts
Comment (MutazAshhab) [reply src/sharedTypes.ts]: enum refactor here: https://github.com/patrickbrett/mlvet/pull/38/commits/f54cbe3641987ce328de2f5e4091eb81a6eb50d4

File: src/renderer/util.ts
Comment (MutazAshhab) [reply src/renderer/util.ts:23]: enum refactor here: https://github.com/patrickbrett/mlvet/pull/38/commits/f54cbe3641987ce328de2f5e4091eb81a6eb50d4

I chose to keep getOS since we might need it down the line

File: src/main/util.ts
Comment (MutazAshhab) [reply src/main/util.ts]: nice catch

enum refactor: https://github.com/patrickbrett/mlvet/pull/38/commits/f54cbe3641987ce328de2f5e4091eb81a6eb50d4

**Comment by emmapainter** [comment]
One last thing (sorry) before I approve! Could you line up the right edge of the x button with the right side of the other elements on the modal for both pages?
Screen Shot 2022-05-07 at 7 27 31 pm

**Comment by MutazAshhab** [comment]
When i remove the right padding it inches closer to where the line you drew is. 

This is the closest it goes since the icon itself has empty space around it.

image

image

Is this acceptable?

**Comment by emmapainter** [comment]
Oh wow I forgot that the icon had padding inside it - I hate that but it's MUI's fault so I guess it's too bad. Love your work, approving now!

**Review by emmapainter — APPROVED** [review]

File: src/renderer/components/NewProjectModal/ImportMediaView.tsx
Comment (ghost) [inline src/renderer/components/NewProjectModal/ImportMediaView.tsx]: Love the stacks. Flexbox convert. Looks like this stack (which displays the uploaded media) is a little bit big and its pushing the back/transcribe buttons down. Maybe reduce the height a bit. 

image

Being aware of multiple files in the future or even just ultra-long filenames now, could be a good idea to add overflow-y: 'auto' (overflowY ?) to this stack. This will add a scrollbar when the content overflows out of the designated height for the container element.

File: src/renderer/components/NewProjectModal/ImportMediaView.tsx
Comment (ghost) [inline src/renderer/components/NewProjectModal/ImportMediaView.tsx]: Rather than set width 40% on both the cancel button and the transcribe button. I suggest you set the width of both to be 100% and then set the gap property (CSS) of the container element to be 32px.

This means that both buttons will try and take up as much space in their parent component as possible (so will be equal) and the gap styling will force the 32px between them.

Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/ImportMediaView.tsx]: Done in this commit: https://github.com/patrickbrett/mlvet/pull/38/commits/db258538dce61ed27b415f8350b7af96a08cf2f0

I also went ahead and adjusted it for `NewProjectView.tsx`

File: src/renderer/components/NewProjectModal/ImportMediaView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/ImportMediaView.tsx]: Nice catch, I went ahead and adjusted the hieght such that it lines up. I also removed `marign-top` in other components.

Most importantly i also added the `overflowY: 'auto'` in this commit: https://github.com/patrickbrett/mlvet/pull/38/commits/eaf8056a93593b9f6945a1618fecd871dbe21670

File: src/renderer/pages/Project.tsx
Comment (ghost) [reply src/renderer/pages/Project.tsx]: Great!

**Review by ghost — APPROVED** [review]

File: src/main/util.ts
Comment (chloebrett) [inline src/main/util.ts:47]: Nitpick - could just map OS identifier strings to enum values and return that, with null as a default fallback

File: src/renderer/components/NewProjectModal/RunTranscriptionView.tsx
Comment (chloebrett) [inline src/renderer/components/NewProjectModal/RunTranscriptionView.tsx]: I don't think 'filePath' is an attribute in 'project' any more - we have 'mediaFilePath' and 'saveFilePath'

File: src/sharedTypes.ts
Comment (chloebrett) [inline src/sharedTypes.ts:46]: This will store the values as numbers, can we instead do like Linux = 'LINUX' etc so the runtime values are human readable

Also, maybe say MacOS instead of Darwin, Darwin is just the string identifier and not many people are familiar with it

**Review by chloebrett — APPROVED** [review]
Couple of super small things, approved once those are done

File: src/renderer/components/NewProjectModal/RunTranscriptionView.tsx
Comment (MutazAshhab) [reply src/renderer/components/NewProjectModal/RunTranscriptionView.tsx]: refactor done here: https://github.com/patrickbrett/mlvet/pull/38/commits/5d57bfca861469a5898825e6172929ba38319e0c

File: src/sharedTypes.ts
Comment (MutazAshhab) [reply src/sharedTypes.ts:46]: human readable refactor + macOS instead of darwin done here: https://github.com/patrickbrett/mlvet/pull/38/commits/d0e83b46c50d1224dbed8e815b067fd19d3ae308

File: src/main/util.ts
Comment (MutazAshhab) [reply src/main/util.ts:47]: Mapped enum valies with the corresponding OS identifiers here: https://github.com/patrickbrett/mlvet/pull/38/commits/d0e83b46c50d1224dbed8e815b067fd19d3ae308

**Comment by MutazAshhab** [comment]
I also refactored `MediaDisplayOnUpload` component and imports to `MediaDisplayOnImport` since @patrickbrett pointed out that the user will be importing not uploading.

File: src/main/util.ts
Comment (chloebrett) [inline src/main/util.ts:53]: This isn't quite what I meant, I was more referring to using a map/object so you don't need three if statements, something like this:

```
const platformToOperatingSystem: Record<string, OperatingSystems> = {
    'darwin': OperatingSystems.MACOS,
    'windows': OperatingSystems.WINDOWS,
    'linux': OperatingSystems.LINUX
};

return platformToOperatingSystem[os.platform()] || null;
```

But - don't worry about it, this is approach is fine. There's a fine line between giving a review and saying "do it how I'd do it". There is nothing wrong with the approach you've taken given the number of options, just consider the object approach if you encounter this problem again with say, 5-10 options as it will cut down on if statements :)

**Review by chloebrett — APPROVED** [review]
Looking gooood! Sorry for holding up your merge for so long hehe

**Pull request merged**
